apps/frontend/src/app/5_pages/AavePage/components/BorrowPositionsList/components/EfficiencyModeCard/components/DisableEModeForm/DisableEModeForm.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { FC, useCallback, useMemo } from 'react';
+
+import { t } from 'i18next';
+
+import {
+  Button,
+  ErrorBadge,
+  ErrorLevel,
+  Icon,
+  IconNames,
+  SimpleTable,
+  SimpleTableRow,
+} from '@sovryn/ui';
+import { Decimal } from '@sovryn/utils';
+
+import { AmountRenderer } from '../../../../../../../../2_molecules/AmountRenderer/AmountRenderer';
+import { config } from '../../../../../../../../../constants/aave';
+import { useAaveSetUserEMode } from '../../../../../../../../../hooks/aave/useAaveSetUserEMode';
+import { useAaveUserReservesData } from '../../../../../../../../../hooks/aave/useAaveUserReservesData';
+import { translations } from '../../../../../../../../../locales/i18n';
+import { EModeCategory } from '../../../../../../../../../types/aave';
+import { CollateralRatioHealthBar } from '../../../../../CollateralRatioHealthBar/CollateralRatioHealthBar';
+import { normalizeEModeSummary } from '../../EfficencyModeCard.utils';
+
+type DisableEModeFormProps = {
+  current: EModeCategory;
+  onComplete: () => void;
+};
+
+export const DisableEModeForm: FC<DisableEModeFormProps> = ({
+  current,
+  onComplete,
+}) => {
+  const { handleDisableUserEMode } = useAaveSetUserEMode();
+  const { userReservesData, reservesData, timestamp } =
+    useAaveUserReservesData();
+
+  const onConfirm = useCallback(() => {
+    handleDisableUserEMode({ onComplete });
+  }, [handleDisableUserEMode, onComplete]);
+
+  const summaryAfterDisabled = useMemo(() => {
+    return normalizeEModeSummary(
+      0, // disabled
+      reservesData,
+      userReservesData,
+      timestamp,
+    );
+  }, [userReservesData, reservesData, timestamp]);
+
+  const confirmEnabled = useMemo(() => {
+    // cannot disable if undercollateralized
+    return !summaryAfterDisabled.liquidationRisk;
+  }, [summaryAfterDisabled.liquidationRisk]);
```

Comment:
I like the renaming. It is clearer but isn't it too much the memo for a boolean assignment? Just to discuss

Reply:
yeah, had the same dilema, I think it can get lost below, and seen they used this in other components, but e true that for one only the memo may sound too much. I can remove it if you think so, no strong position here